src/data-workspace/validation/use-validation-result.js

```diff
@@ -0,0 +1,125 @@
+import { useCallback } from 'react'
+import { useIsMutating, useQueries } from 'react-query'
+import {
+    useDataSetId,
+    useOrgUnitId,
+    usePeriodId,
+} from '../../context-selection/index.js'
+import { useApiAttributeParams } from '../../shared/index.js'
+
+const getValidationMetaDataQueryKey = (datasetId) => {
+    const queryKey = [
+        `validationRules`,
+        {
+            params: {
+                dataSet: datasetId,
+                fields: [
+                    'id',
+                    'importance',
+                    'operator',
+                    'leftSide',
+                    'rightSide',
+                    'displayInstruction',
+                    'displayDescription',
+                    'displayName',
+                ],
+            },
+        },
+    ]
+
+    return queryKey
+}
+
+const getValidationQueryKey = (
+    datasetId,
+    { periodId, orgUnitId, categoryComboId, categoryOptionIds }
+) => {
```

Comment:
I wouldn't separate the data set id from the other options. This probably due to the data set it being a segment in the endpoint path while the others are query parameters, but these are implementation details and should affect the api of this function. Also I'd do `dataSetId` instead of `datasetId`:

```suggestion
const getValidationQueryKey = ({
    dataSetId,
    periodId,
    orgUnitId,
    categoryComboId,
    categoryOptionIds,
}) => {
```